tabular/src/autogluon/tabular/models/tab_transformer/tab_transformer.py

```diff
@@ -0,0 +1,157 @@
+# Code in this script based on work by Milan Cvitkovic,
+# Xin Huang, Ashish Khetan and Zohar Karnin.
+
+import copy
+
+import torch
+import torch.nn as nn
+
+from .tab_model_base import TabModelBase
+
+
+class TabTransformer(TabModelBase):
+    """
+    Transformer model for tabular data, can also be used for semi-supervised learning.
```

Comment:
add clarification to doc, that this transformer is just used to embed the data, with further fully connected layers added on top to produce predictions

Reply:
Added more doc here. Also added a bit more doc in tab_transformer_model to clarify the distinction.